src/NetworkSettings.cpp

```diff
@@ -28,6 +30,27 @@ void NetworkSettingsClass::init(Scheduler& scheduler)
     WiFi.setSortMethod(WIFI_CONNECT_AP_BY_SIGNAL);
 
     WiFi.onEvent(std::bind(&NetworkSettingsClass::NetworkEvent, this, _1));
+
+    if (PinMapping.isValidW5500Config()) {
+        _spiEth = true;
+
+        PinMapping_t& pin = PinMapping.get();
+        if (PinMapping.isValidCmt2300Config() && PinMapping.isValidNrf24Config()) {
+            MessageOutput.println("No ETH connection possible with CMT and NRF enabled.");
+        } else if (PinMapping.isValidCmt2300Config()) {
+#if CONFIG_IDF_TARGET_ESP32S3
+            ETHSPI.begin(pin.w5500_sclk, pin.w5500_mosi, pin.w5500_miso, pin.w5500_cs, pin.w5500_int, pin.w5500_rst, SPI3_HOST);
+#else
+            MessageOutput.println("No SPI3_HOST avialable on current device.");
```

Comment:
Classic ESP32 also has SPI3, right? Also there is a typo here

Reply:
Classic ESP32 Wroom only has HSPI and VSPI, ie two SPI Busses 🚌 according to my memory.

Reply:
> The ESP32 integrates 4 SPI peripherals: SPI0, SPI1, SPI2 (commonly referred to as HSPI), and SPI3 (commonly referred to as VSPI). 

[Source](https://randomnerdtutorials.com/esp32-spi-communication-arduino/)

This code doesn't make use of two SPI buses. If CMT2300A is configured, it just gives up.

And sure enough, when I edit it to use SPI3_HOST it works just fine. On classic ESP32 (I used ESP32 mini)

Reply:
I can NOT confirm this 1 second "delay" while connected via ethernet.

On Linux

`# ping 10.1.140.80 -i 0.2
PING 10.1.140.80 (10.1.140.80) 56(84) bytes of data.
64 bytes from 10.1.140.80: icmp_seq=1 ttl=63 time=1.26 ms
64 bytes from 10.1.140.80: icmp_seq=2 ttl=63 time=2.00 ms
64 bytes from 10.1.140.80: icmp_seq=3 ttl=63 time=1.19 ms
64 bytes from 10.1.140.80: icmp_seq=4 ttl=63 time=1.23 ms
64 bytes from 10.1.140.80: icmp_seq=5 ttl=63 time=2.31 ms
64 bytes from 10.1.140.80: icmp_seq=6 ttl=63 time=2.71 ms
64 bytes from 10.1.140.80: icmp_seq=7 ttl=63 time=1.23 ms
64 bytes from 10.1.140.80: icmp_seq=8 ttl=63 time=1.28 ms
64 bytes from 10.1.140.80: icmp_seq=9 ttl=63 time=1.11 ms
64 bytes from 10.1.140.80: icmp_seq=10 ttl=63 time=1.14 ms
64 bytes from 10.1.140.80: icmp_seq=11 ttl=63 time=2.40 ms
64 bytes from 10.1.140.80: icmp_seq=12 ttl=63 time=1.21 ms
64 bytes from 10.1.140.80: icmp_seq=13 ttl=63 time=1.28 ms
64 bytes from 10.1.140.80: icmp_seq=14 ttl=63 time=1.40 ms
64 bytes from 10.1.140.80: icmp_seq=15 ttl=63 time=1.37 ms
64 bytes from 10.1.140.80: icmp_seq=16 ttl=63 time=1.12 ms
64 bytes from 10.1.140.80: icmp_seq=17 ttl=63 time=1.25 ms
64 bytes from 10.1.140.80: icmp_seq=18 ttl=63 time=1.31 ms
64 bytes from 10.1.140.80: icmp_seq=19 ttl=63 time=1.51 ms
64 bytes from 10.1.140.80: icmp_seq=20 ttl=63 time=1.13 ms
64 bytes from 10.1.140.80: icmp_seq=21 ttl=63 time=1.44 ms
64 bytes from 10.1.140.80: icmp_seq=22 ttl=63 time=1.15 ms
64 bytes from 10.1.140.80: icmp_seq=23 ttl=63 time=1.29 ms
64 bytes from 10.1.140.80: icmp_seq=24 ttl=63 time=1.08 ms
64 bytes from 10.1.140.80: icmp_seq=25 ttl=63 time=1.49 ms
64 bytes from 10.1.140.80: icmp_seq=26 ttl=63 time=2.06 ms
64 bytes from 10.1.140.80: icmp_seq=27 ttl=63 time=1.26 ms
64 bytes from 10.1.140.80: icmp_seq=28 ttl=63 time=1.16 ms
64 bytes from 10.1.140.80: icmp_seq=29 ttl=63 time=1.94 ms
64 bytes from 10.1.140.80: icmp_seq=30 ttl=63 time=1.24 ms
64 bytes from 10.1.140.80: icmp_seq=31 ttl=63 time=2.55 ms
64 bytes from 10.1.140.80: icmp_seq=32 ttl=63 time=2.44 ms
64 bytes from 10.1.140.80: icmp_seq=33 ttl=63 time=1.39 ms
64 bytes from 10.1.140.80: icmp_seq=34 ttl=63 time=1.27 ms
^C
--- 10.1.140.80 ping statistics ---
34 packets transmitted, 34 received, 0% packet loss, time 6618ms
rtt min/avg/max/mdev = 1.078/1.505/2.710/0.469 ms
`

Or Windoze

`C:\>ping 10.1.140.80 -t

Ping wird ausgeführt für 10.1.140.80 mit 32 Bytes Daten:
Antwort von 10.1.140.80: Bytes=32 Zeit=5ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=3ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=5ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=44ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=3ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=3ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=3ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=4ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=3ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=3ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=3ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=3ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=3ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=3ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=3ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=3ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=4ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=3ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=3ms TTL=63
Antwort von 10.1.140.80: Bytes=32 Zeit=6ms TTL=63

Ping-Statistik für 10.1.140.80:
    Pakete: Gesendet = 20, Empfangen = 20, Verloren = 0
    (0% Verlust),
Ca. Zeitangaben in Millisek.:
    Minimum = 3ms, Maximum = 44ms, Mittelwert = 5ms
STRG-C
^C`